libpod/container.go

```diff
@@ -686,12 +685,32 @@ func (c *Container) Init() (err error) {
 	if c.config.CreateNetNS {
 		g.AddOrReplaceLinuxNamespace(spec.NetworkNamespace, c.state.NetNS.Path())
 	}
+	// Remove default /etc/shm mount
+	g.RemoveMount("/dev/shm")
 	// Mount ShmDir from host into container
-	g.AddBindMount(c.config.ShmDir, "/dev/shm", []string{"rw"})
+	mnt := spec.Mount{
+		Type:        "bind",
+		Source:      c.config.ShmDir,
+		Destination: "/dev/shm",
+		Options:     []string{"rw", "bind"},
+	}
+	g.AddMount(mnt)
 	// Bind mount resolv.conf
-	g.AddBindMount(runDirResolv, "/etc/resolv.conf", []string{"rw"})
+	mnt = spec.Mount{
```

Comment:
Can you name this something different? It's a little bit confusing using mnt for 3 different mounts in the same function

Reply:
Done